packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts

```diff
@@ -18,46 +20,171 @@ const useStyles = makeStyles({
+    ...shorthands.borderColor(tokens.colorNeutralStroke1),
+    ...shorthands.borderRadius(tokens.borderRadiusCircular),
+    ...shorthands.borderStyle('solid'),
     ...shorthands.padding(0),
   },
+
+  // These styles match the default button styles
+  focusIndicator: createCustomFocusIndicatorStyle({
+    ...shorthands.borderColor('transparent'),
+    outlineColor: 'transparent',
+    outlineWidth: tokens.strokeWidthThick,
+    outlineStyle: 'solid',
+    boxShadow: `${tokens.shadow4} 0 0 0 2px ${tokens.colorStrokeFocus2}`,
+    zIndex: 1,
+  }),
+
+  states: {
+    ':hover': {
+      color: tokens.colorNeutralForeground1Hover,
+      backgroundColor: tokens.colorNeutralBackground1Hover,
+      ...shorthands.borderColor(tokens.colorNeutralStroke1Hover),
+    },
+    ':active': {
+      color: tokens.colorNeutralForeground1Pressed,
+      backgroundColor: tokens.colorNeutralBackground1Pressed,
+      ...shorthands.borderColor(tokens.colorNeutralStroke1Pressed),
+    },
+  },
+
+  pie: {
+    backgroundColor: tokens.colorTransparentBackground,
+    ...shorthands.borderColor(tokens.colorTransparentStroke),
+    color: 'transparent',
+  },
+
+  icon12: { fontSize: '12px' },
+  icon16: { fontSize: '16px' },
+  icon20: { fontSize: '20px' },
+  icon24: { fontSize: '24px' },
+  icon28: { fontSize: '28px' },
+  icon32: { fontSize: '32px' },
+  icon48: { fontSize: '48px' },
+  caption2Strong: { ...typographyStyles.caption2Strong },
+  caption1Strong: { ...typographyStyles.caption1Strong },
+  body1Strong: { ...typographyStyles.body1Strong },
+  subtitle2: { ...typographyStyles.subtitle2 },
+  subtitle1: { ...typographyStyles.subtitle1 },
+  title3: { ...typographyStyles.title3 },
+  borderThin: { ...shorthands.borderWidth(tokens.strokeWidthThin) },
+  borderThick: { ...shorthands.borderWidth(tokens.strokeWidthThick) },
+  borderThicker: { ...shorthands.borderWidth(tokens.strokeWidthThicker) },
+  borderThickest: { ...shorthands.borderWidth(tokens.strokeWidthThickest) },
+});
+
+/**
+ * Styles for overflow list slot.
+ */
+const useOverflowContentStyles = makeStyles({
+  base: {
+    maxHeight: '220px',
+    minHeight: '80px',
+    ...shorthands.overflow('hidden', 'scroll'),
+    ...shorthands.padding(tokens.spacingHorizontalS),
+    width: '220px',
+  },
 });
 
 /**
  * Apply styling to the AvatarGroup slots based on the state
  */
 export const useAvatarGroupStyles_unstable = (state: AvatarGroupState): AvatarGroupState => {
-  const { size } = state;
+  const { layout, overflowIndicator, size } = state;
   const styles = useStyles();
   const sizeStyles = useSizeStyles();
-  const overflowListStyles = useOverflowListStyles();
+  const overflowContentStyles = useOverflowContentStyles();
+  const overflowButtonStyles = useOverflowButtonStyles();
 
-  state.root.className = mergeClasses(avatarGroupClassNames.root, styles.base, state.root.className);
+  const layoutClassName = useLayoutClassName(layout, size);
```

Comment:
(nit) Maybe call this `useGroupChildClassName`, or something to indicate it applies to the items/elements/children inside the group.